app/dashboard/settings/profile/profileActions.ts

```
@@ -0,0 +1,40 @@
+  // First get the id before validation
+  const { id, ...dataToValidate } = data
+
+  // Validate the update data (excluding id)
+  const validatedData = userSchema.parse(dataToValidate)
+
+  // Convert the data to match Prisma's expected format
+  const updateData: Prisma.UserUpdateInput = Object.entries(
+    validatedData
+  ).reduce(
+    (acc, [key, value]) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)